gatsby-node.js

```diff
@@ -0,0 +1,164 @@
+let resolver;
+let knackPagePromise = new Promise(resolve => {
+  resolver = resolve;
+});
+
+exports.onCreateNode = ({ node }) => {
+  // Required to ensure that the knack MDX is available for sourceNodes
+  if (node.relativePath === 'rulebook/knacks.mdx') {
+    resolver();
+  }
+};
+
+// https://spectrum.chat/gatsby-js/general/why-is-sourcenodes-hook-called-before-all-oncreatenode-callbacks-are-finished~cf2ba3a2-c463-4df0-b691-99c266cf0a43?m=MTU4Mjc0MzYyMDI5Mw==
+exports.sourceNodes = ({ actions, createContentDigest, getNodesByType }) => {
+  const { createNode } = actions;
+  const createKnackNode = knackContent => {
```

Comment:
This would probably be a lot easier to read if knackContent was typed.
I did stumble on this yesterday https://gist.github.com/JohnAlbin/2fc05966624dffb20f4b06b4305280f9 which allows the use of typescript for these files, seems to be a current and at least somewhat widely adopted solution, but there may be other approaches.